components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt

```diff
@@ -15,44 +19,114 @@ import mozilla.components.support.base.log.logger.Logger
  * @param valuesProvider provider for the device's values
  */
 @Suppress("TooManyFunctions")
-class Experiments(
-    private val source: ExperimentSource,
-    private val storage: ExperimentStorage,
-    valuesProvider: ValuesProvider = ValuesProvider()
-) {
+open class ExperimentsInternalAPI internal constructor() {
+    private val logger: Logger = Logger(LOG_TAG)
+
     @Volatile private var experimentsResult: ExperimentsSnapshot = ExperimentsSnapshot(listOf(), null)
     private var experimentsLoaded: Boolean = false
-    private val evaluator = ExperimentEvaluator(valuesProvider)
-    private val logger = Logger(LOG_TAG)
+    private var evaluator: ExperimentEvaluator = ExperimentEvaluator(ValuesProvider())
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var valuesProvider: ValuesProvider = ValuesProvider()
+        set(provider) {
+            field = provider
+            evaluator = ExperimentEvaluator(field)
+        }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var source: ExperimentSource? = null
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var storage: ExperimentStorage? = null
+
+    private var isInitialized = false
+
+    /**
+     * Initialize the experiments library.
+     *
+     * This should only be initialized once by the application.
+     *
+     * @param applicationContext [Context] to access application features, such
+     * as shared preferences.
+     */
+    fun initialize(
+        applicationContext: Context,
+        fetchClient: Client = HttpURLConnectionClient()
+    ) {
+        if (isInitialized) {
+            logger.error("Experiments library should not be initialized multiple times")
+            return
+        }
+
+        experimentsResult = ExperimentsSnapshot(listOf(), null)
+        experimentsLoaded = false
+
+        if (source == null) {
+            source = KintoExperimentSource(
+                EXPERIMENTS_BASE_URL,
+                EXPERIMENTS_BUCKET_NAME,
+                EXPERIMENTS_COLLECTION_NAME,
+                fetchClient
+            )
+        }
+        if (storage == null) {
+            storage = FlatFileExperimentStorage(
+                File(applicationContext.getDir(EXPERIMENTS_DATA_DIR, Context.MODE_PRIVATE),
+                    EXPERIMENTS_JSON_FILENAME)
+            )
+        }
+
+        isInitialized = true
+    }
+
+    /**
+     * Reset the library in tests.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun testReset() {
```

Comment:
nit: This could be marked as `@VisibleForTesting(otherwise = VisibleForTesting.NONE)` to narrow down the visibility even further, since this is a test only function.  I couldn't find any 'private' or local class uses of this that would require the visibility of `PRIVATE`